### ISSUE TITLE
Question: When is this plugin considered "ready"?

### ISSUE THREAD

**Issue by mstoetzer** [ori-edge/k8s_gateway#95]
The CoreDNS /ready endpoint will return HTTP 200, when all plugins are loaded [See this](https://coredns.io/plugins/ready/).

Can this endpoint used to check if all ingress hostnames are loaded into the index?

I want to use the /ready endpoint in a readinessProbe to add this CoreDNS instance to the service endpoints only when the entries are ready. Startup time could be a few seconds and I want to avoid any downtime of the CoreDNS.

**Comment by networkop** [comment]
Yep, this should be fairly easy to add. The `ready` plugin should already be baked into the binary and all that we're missing is a file similar to this one https://github.com/coredns/coredns/blob/5b9b079dabc7f71463cea3f0c6a92f338935039d/plugin/kubernetes/ready.go
Do you want to do a PR @mstoetzer ?
